packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,264 @@
+  loaded: boolean;
+  loadError: any;
+  rowData: CustomData;
+};
+
+const ResourceTable: React.FC<ResourceTableProps> = (props) => {
+  const { t } = useTranslation();
+  const tableColumns = React.useMemo<TableColumn<K8sResourceCommon>[]>(
+    () => [
+      {
+        title: t("Name"),
+        sort: "metadata.name",
+        transforms: [sortable],
+        props: {
+          className: tableColumnInfo[0].className
+        },
+        id: tableColumnInfo[0].id
+      },
+      {
+        title: t("Status"),
+        props: {
+          className: tableColumnInfo[1].className
+        },
+        id: tableColumnInfo[1].id
+      },
+      {
+        title: t("Provider"),
+        props: {
+          className: tableColumnInfo[2].className
+        },
+        id: tableColumnInfo[2].id
+      },
+      {
+        title: t("Buckets"),
+        props: {
+          className: tableColumnInfo[3].className
+        },
+        id: tableColumnInfo[3].id
+      },
+      {
+        title: "",
+        props: {
+          className: tableColumnInfo[4].className
+        },
+        id: tableColumnInfo[4].id
+      }
+    ],
+    [t]
+  );
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null
+  });
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t("Resource Page")}
+      columns={columns}
+      Row={RowRenderer}
+    />
+  );
+};
+
+type CustomData = {
+  launchModal: LaunchModal;
+  kebabActions?: (t: TFunction) => { [key: string]: { value: string } };
+  bucketPolicyMap: { [key: string]: { value: number } };
+};
+
+const RowRenderer: React.FC<RowProps<NamespaceStoreKind, CustomData>> = ({
+  obj,
+  activeColumnIDs,
+  rowData
+}) => {
+  const { t } = useTranslation();
+  const { launchModal, kebabActions, bucketPolicyMap } = rowData;
+  const bucketsCount = bucketPolicyMap[obj?.metadata?.name] || 0;
+  return (
+    <>
+      <TableData {...tableColumnInfo[0]} activeColumnIDs={activeColumnIDs}>
+        {getName(obj)}
+      </TableData>
+      <TableData {...tableColumnInfo[1]} activeColumnIDs={activeColumnIDs}>
+        <OperandStatus operand={obj} />
+      </TableData>
+      <TableData {...tableColumnInfo[2]} activeColumnIDs={activeColumnIDs}>
+        {NOOBAA_PROVIDER_MAP[obj?.spec?.type]}
+      </TableData>
+      <TableData {...tableColumnInfo[3]} activeColumnIDs={activeColumnIDs}>
+        {t("{{bucketsCount}} Buckets", { bucketsCount })}
+      </TableData>
+      <TableData {...tableColumnInfo[4]} activeColumnIDs={activeColumnIDs}>
+        <Kebab
+          launchModal={launchModal}
+          extraProps={{
+            resource: obj,
+            resourceModel: NooBaaNamespaceStoreModel
+          }}
+          customKebabItems={kebabActions}
+        />
+      </TableData>
+    </>
+  );
+};
+
+type GenericListPageProps = {
+  actions?: ModalMap;
+  resourceModel?: K8sModel;
+  kebabActions?: CustomData["kebabActions"];
```

Comment:
In the shared folder under kebab.tsx, export this as a type and use it here

const CustomKebabItemsType  = (t: TFunction) => {
    [key: string]: {
      value: string;
      props?: DropdownItemProps;
    };

Reply:
so it wont break easily

Reply:
https://github.com/red-hat-storage/mcg-ms-console/pull/4/commits/0fd1d02072c13d737d832be920bd176d00e132d3